src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -0,0 +1,159 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Abstractions.Data;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.SqlServer;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.ChangeFeed
+{
+    /// <summary>
+    /// A data store that provides functionalities to get resource changes and resource types for change feed sources.
+    /// </summary>
+    public class SqlServerFhirResourceChangeDataStore : IChangeFeedSource<ResourceChangeData>
```

Comment:
What's the plan for ResourceChangeType?

Reply:
Mapping from ResourceChangeTypeId to ResourceChangeTypeName is unnecessary. Because, ResourceChangeTypeId is mapped to Event Grid type such as Microsoft.HealthcareApis.FhirResourceUpdated using an enum.